addons/hr_holidays/tests/test_accrual_allocations.py

```diff
@@ -4015,24 +4015,138 @@ def test_accrual_allocation_date_in_the_future(self):
+            'carryover_date': 'allocation',
+            'level_ids': level_ids,
         })
-        self.assertEqual(allocation.accrual_plan_id, accrual_plan, "Accrual plan should initially be set.")
 
-        with Form(allocation) as alloc_form:
-            alloc_form.allocation_type = 'regular'
-        self.assertFalse(
-            allocation.accrual_plan_id,
-            "accrual_plan_id should be cleared automatically when type becomes 'regular'."
-        )
-        self.assertEqual(accrual_plan.employees_count, 0, "Accrual plan should not have any linked employees.")
+    def _test_accrual_plan_start_carryover_expiring_3_months(self, second_lvl_6m):
+        """
+            Assert that days expires correctly for a monthly accrual plan granting days
+            at the start of the month (with a carryover validity of 3 months).
+
+            - Create an accrual plan:
+                - Carryover date: allocation start
+                - One level:
+                    - Accrues 2 days monthly at the start of each month
+                    - Carryover policy: all days carry over
+                    - Carried over days validity: 3 months.
+            - Create an allocation that uses the above accrual plan on 23/09/2025:
+                - Starts on 01/07/2024
+
+            Expected behavior: As the allocation was created on 23/09/2025, 24 days should've
+                been accrued from 01/07/2024 to 30/06/2025 (it will expire on 01/10/2025).
+        """
+        with freeze_time('2025-09-23'):
+            accrued_days = 2
+            accrual_plan = self._get_accrual_plan_sample0(accrued_days, 'start', second_lvl_6m)
+            leave_type_day = self._get_leave_type_day_sample1()
+            allocation = self._get_allocation_sample(leave_type_day, accrual_plan, '2025-07-01')
+            allocation.action_approve()
+
+        assertions = [
+            # 13 months accruals
+            ('2026-07-01', 13 * accrued_days, 24),
+            # The day before 24 days should expire
+            ('2026-09-30', 15 * accrued_days, 24),
+            # Carryover expires + new monthly accrual
+            ('2026-10-01', 16 * accrued_days - 24, 0),
+            # Accrual keeps going
+            ('2026-11-01', 17 * accrued_days - 24, 0),
+        ]
+
+        for test_date, remaining_leaves, expiring_days in assertions:
+            with freeze_time(test_date):
+                assert_virtual_leaves_equal(self, test_date, allocation, leave_type_day, remaining_leaves, self.employee_emp, digits=2)
+                self.assertAlmostEqual(allocation.expiring_carryover_days, expiring_days, 2, msg=f'Incorrect number of expiring days for {test_date}')
+
+    def test_accrual_plan_start_carryover_expiring_3_months1(self):
+        self._test_accrual_plan_start_carryover_expiring_3_months(False)
+
+    def test_accrual_plan_start_carryover_expiring_3_months2(self):
+        self._test_accrual_plan_start_carryover_expiring_3_months(True)
```

Comment:
I don't really see the point of doing the exact same test twice with a nuance on which level we're at 🤔 To make the tests more clear and readable, just make one with an accrual plan with a level for the first two years and a second level after that